### PR TITLE
Use BlockFormEvent for mud converting into clay

### LINKED ISSUE #8366: No way to determine if mud changed to clay
### Expected behavior

It would be useful to have an event where you could get the initial block ( mud ) and the final block ( clay ).

### Observed/Actual behavior

I only found the BlockPhysicsEvent triggered but i didn't find any way to get the initial block ( mud ).

### Steps/models to reproduce

Create the appropriate structure to create some clay from mud 

### Plugin and Datapack List

/

### Paper version

Paper version git-Paper-153 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 63cb747)

### Other

Here is the structure that changes the mud to clay if you wait a bit ( or just increase the randomTickSpeed )
Screenshot_1

### PR DESCRIPTION
**Pull request by Remynfv**
Fixes #8366

Very small change to make mud converting into clay call a BlockFormEvent.

This is my first contribution! Hopefully I got everything right. I tried to keep consistent with the style of the surrounding code.

### CODE REVIEW

File: patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                                 BlockState iblockdata1 = Blocks.CLAY.defaultBlockState();
+ 
+-                                world.setBlockAndUpdate(((PointedDripstoneBlock.FluidInfo) optional.get()).pos, iblockdata1);
+-                                Block.pushEntitiesUp(((PointedDripstoneBlock.FluidInfo) optional.get()).sourceState, iblockdata1, world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos);
+-                                world.gameEvent(GameEvent.BLOCK_CHANGE, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos, GameEvent.Context.of(iblockdata1));
+-                                world.levelEvent(1504, blockposition1, 0);
++                                // Paper start
++                                if (org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockFormEvent(world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos, iblockdata1)) {
++                                    Block.pushEntitiesUp(((PointedDripstoneBlock.FluidInfo) optional.get()).sourceState, iblockdata1, world, ((PointedDripstoneBlock.FluidInfo) optional.get()).pos);
```
Comment (electronicboy) [inline patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch]: Generally, refain from changing the indentation of vanilla code where tenable, just helps reduce diff/merge noise

**Review by electronicboy — APPROVED** [review]
1 trivial change, but, otherwise, lgtm;

File: patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch (hunk 1 above)
Comment (Remynfv) [reply patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch]: Noted, thanks

Comment (Owen1212055) [reply patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch]: (if you could revert the indentation changes and modify the patch that would be great!)

Comment (Remynfv) [reply patches/server/0918-use-BlockFormEvent-for-mud-converting-into-clay.patch]: Amended the commit. Indentation should be fixed now if I did this correctly!

**Review by me4502 — APPROVED** [review]

**Pull request merged**